crates/wasmparser/src/validator/operators.rs

```diff
@@ -625,6 +617,37 @@ where
         }
     }
 
+    /// Pop a reference type from the operand stack, checking if it is a subtype
+    /// of `expected` or the shared version of `expected`. This function will
+    /// panic if `expected` is shared or a concrete type.
+    fn pop_maybe_shared_ref(&mut self, expected: RefType) -> Result<Option<RefType>> {
```

Comment:
To avoid redudant calls to `is_shared_ref_type` could this return `Result<Option<(RefType, bool)>>` where the `bool` is whether it's shared or not?